CoffeeChat/SceneDelegate.swift

```diff
@@ -12,42 +12,56 @@ import UIKit
         // Use this method to optionally configure and attach the UIWindow `window` to the provided UIWindowScene `scene`.
         // If using a storyboard, the `window` property will automatically be initialized and attached to the scene.
-        // This delegate does not imply the connecting scene or session are new
-        // (see `application:configurationForConnectingSceneSession` instead).
+        // This delegate does not imply the connecting scene or session are new (see `application:configurationForConnectingSceneSession` instead).
         guard let scene = scene as? UIWindowScene else { return }
 
         IQKeyboardManager.shared.enable = true
         IQKeyboardManager.shared.keyboardDistanceFromTextField = 200 // TODO: Double check with design
 
         let window = UIWindow(windowScene: scene)
-        guard let signIn = GIDSignIn.sharedInstance() else {
-            let navigationController = UINavigationController(rootViewController: LoginViewController())
-            window.rootViewController = navigationController
-            return
-        }
-        if signIn.hasPreviousSignIn() {
+        let navigationController = UINavigationController(rootViewController: LoginViewController())
+        if let signIn = GIDSignIn.sharedInstance(), signIn.hasPreviousSignIn() {
             signIn.restorePreviousSignIn()
             // Onboard user if they haven't done so yet, otherwise bring to home.
-            let onboardingCompleted = userDefaults.bool(forKey: Constants.UserDefaults.onboardingCompletion)
+            let onboardingCompleted = UserDefaults.standard.bool(forKey: Constants.UserDefaults.onboardingCompletion)
+            let refreshToken = UserDefaults.standard.string(forKey: Constants.UserDefaults.refreshToken)
+            let assignedMatch = true
             let homeVC = HomeViewController()
-            let onboardingVC = OnboardingPageViewController(
-                transitionStyle: UIPageViewController.TransitionStyle.scroll,
-                navigationOrientation: UIPageViewController.NavigationOrientation.horizontal
-            )
-            let rootVC = onboardingCompleted ? homeVC : onboardingVC
-            let navigationController = UINavigationController(rootViewController: rootVC)
-            window.rootViewController = navigationController
-        } else {
-            // Ask user to sign in if they have not signed in before.
-            let navigationController = UINavigationController(rootViewController: LoginViewController())
-            window.rootViewController = navigationController
+            let noMatchVC = NoMatchViewController()
+            let onboardingVC = OnboardingPageViewController(transitionStyle: .scroll, navigationOrientation: .horizontal)
+            let matchVC = assignedMatch ? homeVC : noMatchVC
+            let rootVC = onboardingCompleted ? matchVC : onboardingVC
```

Comment:
more so a stylistic preference thing, but i feel like saving the VCs into constants and then using the constants in the ternary doesn't help readability that much but adds quite a few lines -- especially for `homeVC` and `noMatchVC` where the classes themselves (e.g. `HomeViewController`) are descriptive already. up to you though!